frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal/tests/setup.tsx

```diff
@@ -0,0 +1,101 @@
+/* istanbul ignore file */
+import { useState } from "react";
+import {
+  createMockSearchResult,
+  createMockTokenFeatures,
+} from "metabase-types/api/mocks";
+import { SearchModelType, TokenFeatures } from "metabase-types/api";
+import { mockSettings } from "__support__/settings";
+import { createMockState } from "metabase-types/store/mocks";
+import { setupEnterprisePlugins } from "__support__/enterprise";
+import { renderWithProviders, screen, waitFor } from "__support__/ui";
+import { SearchFilterModal } from "metabase/search/components/SearchFilterModal/SearchFilterModal/SearchFilterModal";
+import { setupSearchEndpoints } from "__support__/server-mocks";
+import { SearchFilters } from "metabase/search/types";
+
+export interface SetupOpts {
+  tokenFeatures?: TokenFeatures;
+  hasEnterprisePlugins?: boolean;
+  initialFilters?: SearchFilters;
+  availableModelTypes?: Array<SearchModelType>;
+}
+
+export const TEST_TYPES: Array<SearchModelType> = [
+  "card",
+  "collection",
+  "dashboard",
+  "database",
+  "dataset",
+  "metric",
+  "pulse",
+  "segment",
+  "table",
+];
+
+export const COMMON_INITIAL_FILTERS: SearchFilters = {
+  type: TEST_TYPES,
+};
+
+export const PREMIUM_INITIAL_FILTERS: SearchFilters = {
+  ...COMMON_INITIAL_FILTERS,
+  verified: true,
+};
+
+const TestSearchFilterModal = ({
+  initialFilters = {},
+  onChangeFilters,
+}: {
+  initialFilters?: SearchFilters;
+  onChangeFilters: jest.Mock;
+}) => {
+  const [filters, setFilters] = useState(initialFilters);
+
+  onChangeFilters.mockImplementation(newFilters => setFilters(newFilters));
```

Comment:
can't we simply wrap this in another function rather than mocking the implementation? This looks too complex.

e.g.
```suggestion
  function handleChangeFilters(newFilters) {
    setFilters(newFilters)
    onChangeFilter(newFilters)
  }
```

Reply:
Yeah, I was back and forth on this one but my thinking was that I wanted to use as much jest as possible. I'll change it back, I have no real preferences here